### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Nessrine88**
In this milestone I have :
- Added a column species of type string to animals table.
createColumn
- Inserted the following data:
add data to rows
- updated the animals table by setting the species column to unspecified
unspecified
- rolling back the change
ROLLBACK
- Updated the animals table by setting the species column to digimon for all animals that have a name ending in mon:
set species to digimon
- Updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
POKEMONsPECIES
- deleted all records in the animals table, then roll back the transaction:
delete
- Deleted all animals born after Jan 1st, 2022.
DELETE WITH CONDITION
- Created a savepoint for the transaction.
- Updated all animals' weight to be their weight multiplied by -1.
- Updated all animals' weights that are negative to be their weight multiplied by -1.
update weight
ROLLBACK TO THE SAVEPOINT
make all weights positive

**Write queries to answer the following questions:**

- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- 
COUNT
number of animal never escaped

average weight

max aescape attempts
min and max group
avg of escape for specific date for type of animals

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+CREATE TABLE animals(
+    id INTEGER,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL,
+    escape_attempts INTEGER NOT NULL,
+    neutered BOOLEAN NOT NULL,
+    weight_kg DECIMAL NOT NULL,
+    species VARCHAR(100),
+    PRIMARY KEY(id)
+);
```
Comment (V-Blaze) [inline schema.sql]: - You have done well so far :+1: :+1: But there are a few changes and requirements you need to meet before moving to the next milestone. :+1: :waffle: 

- [ ] When working with databases, you don't edit an already existing file to add or remove a column in a table, you add a line of SQL to do that. In this case, you are not meant to edit the animal table and add the `species` column, you rather need to do it this way.
```sh
ALTER TABLE animals ADD species varchar(100);
```
I'll suggest you update your schema file accordingly. :wave: :smile:

File: data.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+('Gabumon', '2018-11-15', 2, TRUE, 8),
+('Pikachu', '2021-01-07', 1, FALSE, 15.04),
+('Devimon','2017-05-12', 5, TRUE, 11),
+('Charmander', '2020-02-08', 0, FALSE, -11),
+('Plantmon', '2021-11-15', 2, TRUE, -5),
+('Squirtle', '1993-04-02', 3, TRUE, -12.13),
+('Angemon', '2005-06-12', 1, TRUE, -45),
+('Boarmon', '2005-06-17', 7, TRUE, 20.4),
+('Blossom', '1998-10-13', 3, TRUE, 17),
+('Ditto', '2022-04-14', 4, TRUE, 22);
```
Comment (V-Blaze) [inline data.sql]: -  [ ] For this Milestone and subsequent milestones, I suggest you separate the task from the one in the previous milestone, it makes your code easier to follow and understand..

- [ ] The data you filled in here `('Plantmon', '2021-11-15', 2, TRUE, -5),` is not fully correct. According to the requirement, the weight is `-5.7`. I'll suggest you update this accordingly. :+1: :100: 
- [ ] Also the data you have here `('Boarmon', '2005-06-17', 7, TRUE, 20.4),` is not fully correct. The correct `date_of_birth` should be `2005-06-07`. I'll suggest you update this. :+1: :wave: 
- [ ] Also the data here `('Squirtle', '1993-04-02', 3, TRUE, -12.13),` is not fully correct. `neutered` should be `FALSE` according to the requirement. I'll suggest you update this. :+1: :airplane:

File: queries.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
+SELECT * FROM animals WHERE name LIKE '%mon';
+SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
+SELECT name FROM animals WHERE neutered = TRUE AND escape_attempts < 3;
+SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+SELECT name FROM animals WHERE neutered = TRUE;
+SELECT name FROM animals WHERE name!='Gabumon';
+SELECT name FROM animals WHERE weight_kg>=10.5 AND weight_kg <= 17.3;
+BEGIN;
```
Comment (V-Blaze) [inline queries.sql]: - [ ] You have done well writing out the queries but Looking at the requirement, you are missing the first 3 queries.
- - Screenshot reference:
Selection_009
- [ ] I'll suggest you go back to the requirement, reed the instructions, and write out these queries. :100: :1st_place_medal: 

Also, remember to update the screenshot accordingly :smile: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Nessrine88 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct Gitflow✔
- Most queries written correctly 👋 
-  Some data inserted correctly into the database ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Nessrine88** [comment]
Changes Requested:
image

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Nessrine88, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good SQL query design :white_check_mark: 
- [x] Animals table created and updated correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**